Client/Coordinators/Browser/BrowserCoordinator.swift

```diff
@@ -339,10 +339,15 @@ class BrowserCoordinator: BaseCoordinator,
 
     // MARK: - LibraryCoordinatorDelegate
 
-    func didFinishLibrary(from coordinator: LibraryCoordinator) {
+    func didFinishLibrary(from coordinator: Coordinator) {
         router.dismiss(animated: true, completion: nil)
-        remove(child: coordinator)
+        if !(coordinator is LibraryCoordinator), let libraryCoordinator = childCoordinators.first(where: { $0 is LibraryCoordinator }) as? LibraryCoordinator {
+            remove(child: libraryCoordinator)
```

Comment:
Why in some cases we need to cast to LibraryCoordinator? Else case would not work?

Reply:
i did this beacuse when i call did finish on the bookmark coordinator it also closes the library view controller so with the previous implementation it was just removing the bookmark coordinator and not the whole library coordinator. Anyway with your review, this implementation will be not needed anymore because the bookmark coordinator will not handle the open of url and then the did finish

Reply:
Ok thanks for the explanation!

Reply:
You re welcome